cli/root_test.go

```diff
@@ -28,7 +27,10 @@ func resetCobra(cmd *cobra.Command) {
 	}
 }
 
-func execute(cmd *cobra.Command, args ...string) (string, error) {
+func execute(t *testing.T, cmd *cobra.Command, args ...string) (string, error) {
```

Comment:
This could be improved by using `t.Cleanup()` to reset Cobra after the test invocation finishes. I know we reset before each test invocation as well.

```
t.Cleanup(func() {
  resetCobra(t, cmd)
})
```

Reply:
Additionally, instead of this function returning `(string, error)`, consider having the function immediately exit on a failure condition - this means the callee doesn't have to check as much:

```go
var buf bytes.Buffer
cmd.SetOutput(&buf)
if err := cmd.Execute(); err != nil {
    t.Errof("unexpected error: %s", err)
    t.FailNow()
}
return buf.String()
```

Reply:
Thanks for the feedback,  You are certainly teaching me a few tricks.  I would love to take a stab at the suggestions you are making as a learning opportunity.  Super busy this week, but might find some time in the next week or so to see what I can come up with.